src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/ReadMePreviewViewModel.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Telemetry;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public sealed class ReadMePreviewViewModel : ViewModelBase
+    {
+        public ReadMePreviewViewModel()
+        {
+            _isErrorWithReadMe = false;
+            _rawReadMe = string.Empty;
+        }
+
+        private bool _isErrorWithReadMe = false;
+
+        public bool IsErrorWithReadMe
+        {
+            get => _isErrorWithReadMe;
+            set
+            {
+                if (_isErrorWithReadMe != value)
+                {
+                    _isErrorWithReadMe = value;
+                    RaisePropertyChanged(nameof(IsErrorWithReadMe));
+                }
+            }
+        }
+
+        private string _rawReadMe = string.Empty;
+
+        public string ReadMeMarkdown
```

Comment:
Is it possible to stream the readme content instead of storing the entire content in a string? The idea is to retrieve the stream based on the view port and extract more information if the user scrolls down. I guess this approach would be memory efficient especially for large readme files.

Reply:
Hmm, currently the control only takes the raw string so I don't think it's possible to use a stream. Do we see instances of very large readmes? This would also impact the package size right?

Reply:
Given that packages are zip achieve, the compressed size depends upon the data in the readme file. The uncompressed file size is not a concern because it is stored in the global packages folder. 

AFAFIK, if the string has around 85 kb runtime allocate that string in large object heap. I assume it is also rare for a readme file to have 85 kb. Maybe we could create a work item for the control author and plan next steps.

Reply:
@joelverhagen is it possible to get information on how big README actually get?

Reply:
`HEAD` request on the blob URL works, if it's hosted on blob storage. This returns `Content-Length` (size in bytes). If you want to do ad hoc analysis, we have our NuGet Insights Kusto cluster so you can check all READMEs on NuGet.org.